libc/baselibc/src/memcmp.c

```diff
@@ -6,14 +6,47 @@
 
 int memcmp(const void *s1, const void *s2, size_t n)
 {
+    int d = 0;
+
+#if defined(ARCH_cortex_m3) || defined(ARCH_cortex_m4) || defined(ARCH_cortex_m7)
+    asm (".syntax unified                   \n"
+         "       push {r4, r5, r6}          \n"
+         "       mov  r5, #0                \n"
+         "       and  r6, r2, #0xfffffffc   \n"
+         "       b    test1                 \n"
+         "loop1: ldr  r3, [r0, r5]          \n"
+         "       ldr  r4, [r1, r5]          \n"
+         "       cmp  r3, r4                \n"
+         "       bne  loop2                 \n"
+         "       add  r5, #4                \n"
+         "test1: cmp  r5, r6                \n"
+         "       bne  loop1                 \n"
+         "       b    test2                 \n"
+         "res1:  rev  r3, r3                \n"
```

Comment:
I am not the most conversant with inline assembly but how does the code ever get to the res1 label?

Reply:
Good catch - it should jump here from line 20. This however still works since it jumps to `loop2` instead which compares word byte by byte. I added `res1` label later and forgot to update branch :-)